src/pykeen/models/base.py

```diff
@@ -319,18 +312,14 @@ def can_slice_t(self) -> bool:
         return _can_slice(self.score_t)
 
     @property
-    def modules_not_supporting_sub_batching(self) -> Collection[nn.Module]:
+    def modules_not_supporting_sub_batching(self) -> Iterable[nn.Module]:
         """Return all modules not supporting sub-batching."""
-        return [
-            module
-            for module in self.modules()
-            if isinstance(module, UNSUPPORTED_FOR_SUBBATCHING)
-        ]
+        return get_batchnorm_modules(base=self)
 
     @property
     def supports_subbatching(self) -> bool:  # noqa: D400, D401
         """Does this model support sub-batching?"""
-        return len(self.modules_not_supporting_sub_batching) == 0
+        return empty(self.modules_not_supporting_sub_batching)
```

Comment:
the `empty()` function is a bit redundant - collections automatically have a truthy/falsy evaluation based on whether they're empty or not

```suggestion
        return not self.modules_not_supporting_sub_batching
```

Will do the job just fine :)

Reply:
but this is not a collection but an iterable

Reply:
I made a comment below, because this won't work on an iterable. However, I think it makes more sense for this to be a collection and not just a generator

Reply:
also, the `empty()` function will consume the iterable either way, so wouldn't it be easier to consume it using the list() operation then work with the list? It's not going to ever be so big

Reply:
Resolved in 2ba1e362